lib/facter/self_service.rb

```diff
@@ -114,6 +114,81 @@
     { S0014: res.nil? }
   end
 
+  chunk(:S0016) do
+    # Puppetserver
+    next unless PuppetSelfService.primary? || PuppetSelfService.compiler? || PuppetSelfService.legacy_compiler?
+    log_path = Puppet.settings['logdir'].to_s + '/../puppetserver/puppetserver.log'
+    line_count = `wc -l < #{log_path}`.to_i
```

Comment:
Since we are already shelling out to get the number of lines, why not just do a `tail -n 250 <file> | grep java.lang.OutOfMemoryError` for this check?